### PR TITLE
SmartCharging: Implement K01.FR.26

### PR DESCRIPTION
**Pull request by christopher-davis-afs**
## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have made corresponding changes to the documentation
- [x] If OCPP 2.0.1: I have updated the [OCPP 2.0.1 status document](https://github.com/EVerest/libocpp/tree/main/doc/ocpp_201_status.md)
- [x] I read the [contribution documentation](https://github.com/EVerest/EVerest/blob/main/CONTRIBUTING.md) and made sure that my changes meet its requirements

### CODE REVIEW

File: tests/lib/ocpp/v201/test_smart_charging_handler.cpp [hunk 1]
```diff
@@ -283,6 +289,23 @@ TEST_F(ChargepointTestFixtureV201, K01FR19_NumberPhasesOtherThan1AndPhaseToUseSe
     EXPECT_THAT(sut, testing::Eq(ProfileValidationResultEnum::ChargingSchedulePeriodInvalidPhaseToUse));
 }
 
+TEST_F(ChargepointTestFixtureV201,
+       K01FR26_IfChargingRateUnitIsNotInChargingScheduleChargingRateUnits_ThenProfileIsInvalid) {
```
Comment (gberardi-pillar) [inline tests/lib/ocpp/v201/test_smart_charging_handler.cpp:293]: Should we also have the positive/valid cases?

File: lib/ocpp/v201/smart_charging.cpp [hunk 2]
```diff
@@ -236,6 +238,14 @@ ProfileValidationResultEnum
...
                                                  std::optional<EvseInterface*> evse_opt) const {
     for (auto& schedule : profile.chargingSchedule) {
+        // K01.FR.26; We currently need to do string conversions for this manually because our DeviceModel class does
+        // not let us get a vector of ChargingScheduleChargingRateUnits.
+        auto supported_charging_rate_units =
+            this->device_model->get_value<std::string>(ControllerComponentVariables::ChargingScheduleChargingRateUnit);
+        if (supported_charging_rate_units.find(conversions::charging_rate_unit_enum_to_string(
+                schedule.chargingRateUnit)) == supported_charging_rate_units.npos) {
+            return ProfileValidationResultEnum::ChargingScheduleChargingRateUnitUnsupported;
+        }
```
Comment (ghost) [inline lib/ocpp/v201/smart_charging.cpp:248]: Could the conversion be it own helper method? That way you can test the functionality.

File: doc/ocpp_201_status.md [hunk 3]
```diff
@@ -1160,7 +1160,7 @@ This document contains the status of which OCPP 2.0.1 numbered requirements have
 | K01.FR.20 | :white_check_mark: |  |
```
Comment (ghost) [inline doc/ocpp_201_status.md]: I know it is not specific to this change but can you uncheck FR.20 and add a Remark?

**Review by gberardi-pillar — APPROVED** [review]

**Review by ghost — APPROVED** [review]
LGTM - ☕

**Pull request merged**